h5pt.go

```diff
@@ -118,11 +119,27 @@ func (t *Table) ReadPackets(start, nrecords int, data interface{}) error {
 // Struct values must only have exported fields, otherwise extractValues will
 // panic.
 func extractValues(ad *appendData, data interface{}) error {
+	if data, ok := data.(interface{ CMarshal() ([]byte, error) }); ok {
+		raw, err := data.CMarshal()
+		if err != nil {
+			return err
+		}
+
+		tmp := make([]byte, ad.offset+len(raw))
+		copy(tmp, ad.cBuffer)
+		tmp = ad.cBuffer
+		copy(ad.cBuffer[ad.offset:], raw)
+
+		return nil
+	}
+
 	rv := reflect.Indirect(reflect.ValueOf(data))
 	rt := rv.Type()
 
-	var dataPtr unsafe.Pointer
-	var dataSize int
+	// We will at most append 8 bytes at one time.
+	tmp := make([]byte, ad.offset+8)
```

Comment:
I am not sure what we try to do here...
is this done to ensure we have at least `ad.offset+8` of capacity ?

Reply:
ah. ok. now I get it.

a perhaps clearer way (at least to me, YMMV) to ensure we have the correct "padding" (or alignment) would be:
```go
ad.cbuf = append(ad.cbuf, make([]byte, padding)...)
```
where padding is some value we correctly compute :)

Reply:
Yes. It's to ensure that the padding is correct.

Yeah your way would make that cleaner. Maybe I will wrap it in an if-statement so that we just run it whenever `padding > 0` and `padding := ad.offset - len(ad.cBuffer)` .

Will fix tmrw.

Reply:
SGTM.